app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -749,31 +756,42 @@ class MainViewModel @Inject constructor(
         threadsUids: List<String>,
         messageUid: String? = null,
     ) = viewModelScope.launch(ioCoroutineContext) {
-        val mailbox = currentMailbox.value!!
-        val destinationFolder = folderController.getFolder(destinationFolderId)!!
         val threads = threadController.getThreads(threadsUids).ifEmpty { return@launch }
         val message = messageUid?.let { messageController.getMessage(it)!! }
+        val destinationFolder = folderController.getFolder(destinationFolderId)!!
+        val messagesToMove = sharedUtils.getMessagesToMove(threads, message)
 
-        val messages = sharedUtils.getMessagesToMove(threads, message)
+        moveThreadsOrMessageTo(destinationFolder, threads, message, messagesToMove)
+    }
+
+    suspend fun moveThreadsOrMessageTo(
+        destinationFolder: Folder,
+        threads: List<Thread>,
+        message: Message? = null,
+        messagesToMove: List<Message>,
+        shouldDisplaySnackbar: Boolean = true,
+    ) {
+        val mailbox = currentMailbox.value!!
+        val threadsUids = threads.map { it.uid }
 
         threadController.updateIsLocallyMovedOutStatus(threadsUids, hasBeenMovedOut = true)
 
-        val apiResponses = ApiRepository.moveMessages(mailbox.uuid, messages.getUids(), destinationFolder.id)
+        val apiResponses = ApiRepository.moveMessages(mailbox.uuid, messagesToMove.getUids(), destinationFolder.id)
 
         if (apiResponses.atLeastOneSucceeded()) {
             if (shouldAutoAdvance(message, threadsUids)) autoAdvanceThreadsUids.postValue(threadsUids)
 
             refreshFoldersAsync(
                 mailbox = mailbox,
-                messagesFoldersIds = messages.getFoldersIds(exception = destinationFolder.id),
+                messagesFoldersIds = messagesToMove.getFoldersIds(exception = destinationFolder.id),
                 destinationFolderId = destinationFolder.id,
                 callbacks = RefreshCallbacks(onStart = ::onDownloadStart, onStop = { onDownloadStop(threadsUids) }),
             )
         }
 
         threadController.updateIsLocallyMovedOutStatus(threadsUids, hasBeenMovedOut = false)
```

Comment:
It looks like this line has been added in the `archive` case.

Reply:
Also, it looks like this line has been moved out of the `else` block in the `spam` case.